sopel/cli/run.py

```diff
@@ -0,0 +1,619 @@
+        pid_file.write(str(os.getpid()))
+
+    # Step Four: Run Sopel
+    ret = run(config_module, pid_file_path)
+
+    # Step Five: Shutdown Clean-Up
+    os.unlink(pid_file_path)
+
+    if ret == -1:
+        # Restart
+        os.execv(sys.executable, ['python'] + sys.argv)
+    else:
+        # Quit
+        return ret
+
+
+def command_configure(opts):
+    """Sopel Configuration Wizard"""
+    if getattr(opts, 'modules', False):
+        _wizard('mod', opts.config)
+    else:
+        _wizard('all', opts.config)
+
+
+def command_stop(opts):
+    """Stop a running Sopel instance"""
+    # Get Configuration
+    try:
+        settings = utils.load_settings(opts)
+    except ConfigurationNotFound as error:
+        tools.stderr('Configuration "%s" not found' % error.filename)
+        return ERR_CODE
+
+    if settings.core.not_configured:
+        stderr('Sopel is not configured, can\'t stop')
+        return ERR_CODE
+
+    # Redirect Outputs
+    utils.redirect_outputs(settings, False)
+
+    # Get Sopel's PID
+    filename = get_pid_filename(opts, settings.core.pid_dir)
+    pid = get_running_pid(filename)
+
+    if pid is None or not tools.check_pid(pid):
+        stderr('Sopel is not running!')
+        return ERR_CODE
+
+    # Stop Sopel
+    if opts.kill:
+        stderr('Killing the Sopel')
+        os.kill(pid, signal.SIGKILL)
+        return
+
+    stderr('Signaling Sopel to stop gracefully')
+    if hasattr(signal, 'SIGUSR1'):
+        os.kill(pid, signal.SIGUSR1)
+    else:
+        # Windows will not generate SIGTERM itself
+        # https://docs.microsoft.com/en-us/cpp/c-runtime-library/reference/signal
+        os.kill(pid, signal.SIGTERM)
+
+
+def command_restart(opts):
+    """Restart a running Sopel instance"""
+    # Get Configuration
+    try:
+        settings = utils.load_settings(opts)
+    except ConfigurationNotFound as error:
+        tools.stderr('Configuration "%s" not found' % error.filename)
+        return ERR_CODE
+
+    if settings.core.not_configured:
+        stderr('Sopel is not configured, can\'t stop')
+        return ERR_CODE
+
+    # Redirect Outputs
+    utils.redirect_outputs(settings, False)
+
+    # Get Sopel's PID
+    filename = get_pid_filename(opts, settings.core.pid_dir)
+    pid = get_running_pid(filename)
+
+    if pid is None or not tools.check_pid(pid):
+        stderr('Sopel is not running!')
+        return ERR_CODE
+
+    stderr('Asking Sopel to restart')
+    if hasattr(signal, 'SIGUSR2'):
+        os.kill(pid, signal.SIGUSR2)
+    else:
+        # Windows will not generate SIGILL itself
+        # https://docs.microsoft.com/en-us/cpp/c-runtime-library/reference/signal
+        os.kill(pid, signal.SIGILL)
+
+
+def command_legacy(opts):
+    """Legacy Sopel run script
+
+    The ``legacy`` command manages the old-style ``sopel`` command line tool.
+    Most of its features are replaced by the following commands:
+
+    * ``sopel start`` replaces the default behavior (run the bot)
```

Comment:
We probably don't want to _require_ `start`. Running just `sopel` with no arguments at all should always do something useful like running with a default configuration, just as it does now.

The default action should be `start` once the `legacy` parser is removed in (probably) Sopel 8. Any reason not to document it this way immediately? One less thing to forget about later (and goodness knows Sopel's docs/docstrings are already full of obsolete stuff that needs updating…).

Reply:
This is more or less documented here in the `main` function:

        # make sure to have an action first (`legacy` by default)
        # TODO: `start` should be the default in Sopel 8

And this part of the docstring makes it clear enough IMHO that there is more to see:

    .. seealso::

       The github issue `#1471`__ tracks various changes requested for future
       version of Sopel, some of them are related to this legacy command line
       tool.

I would argue that, eventually, the function `command_legacy` will be removed for Sopel 8. This is more visible now than it was at the time of your review, thanks to all the little comments I added after you suggested them.